### DISCUSSION TITLE
Different File Directories after Migrating from UWP to WinUI3

### DISCUSSION THREAD

**Discussion by SprengerS** [microsoft/WindowsAppSDK#4195] [Q&A]
We are using `Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData)` to create our databases since the beginning of our app development with UWP. After migrating to WinUI3 our databases are not visible any more. My investigations has shown  ,that the real file path changed from `LocalState` to `LocalCache`. Are there any ideas how we can avoid this or make our old databases from uwp available for the new WinUI migration?

**Comment by ghost1372** [comment]
MSIX packaged desktop app is not really "sandboxed" it writes to its own virtual registry and application data folder.
if you dont want to disable virtualization, you can find your folder with:
`Windows.Storage.ApplicationData.Current.LocalFolder.Path` 

for UnPackaged app, you can find your app folder in **%AppData%**

**Comment by DarranRowe** [comment]
I think using the UWP functionality would be better here.
[Windows.Storage.ApplicationData](https://learn.microsoft.com/en-us/uwp/api/windows.storage.applicationdata?view=winrt-22621) allows you to obtain the path that you want. ApplicationData.Local allows you to get the LocalState inside the package's application data directory.
As to why this happened, I can't say for sure. The reference source for the .NET framework uses SHGetFolderPath in [Environment.GetFolderPath](https://referencesource.microsoft.com/#mscorlib/system/environment.cs,1477). On the other hand, .NET uses SHGetKnownFolderPath in [Environment.GetFolderPath](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Environment.Win32.cs#L208). You would have to check the output of these to see if they return different paths. It could also be that the UWP version just uses a different method of getting the path which resolves to a different path than what the shell reports.

**Comment by SprengerS** [comment]
The virtualized path from `Environment.GetFolderPath` has changed with the migration from UWP to WinUI from `LocalState` to `LocalCache`. With `Windows.Setup.ApplicationData` we get the old UWP Path. Should we copy the database to the `Environment.GetFolderPath` ?
